core/src/controls.ts

```diff
@@ -74,8 +106,9 @@ export class BinaryCounter extends ControlSignal {
 export class ChordProgression extends ControlSignal {
   readonly encodedChords: tf.Tensor1D[];
 
-  constructor(numSteps: number, chords: string[], encoder: ChordEncoder) {
-    super(numSteps, encoder.depth);
+  constructor(args: ChordProgressionArgs, chords: string[]) {
```

Comment:
I was thinking that the constructor would only take the args so each model would have a single ControlSignal object for its lifetime. Then on each call, you would pass in an arg (e.g., a chord string) to generate a list of tensors for all steps, rather than calling a method for each step. This would be in line with how the DataConverter works.

Reply:
But what type should that arg be?  It works out in DataConverter as it always operates on a NoteSequence, but a) it's weird to have an extra function call for BinaryCounter that provides no information, and b) the fact is, different control signals cannot be treated equivalently as different DataConverters can (from the model's point of view); some control signals need user input, and some don't.

Reply:
My thinking was that you would have a method like  `getTensors(args:ControlSequenceArgs)`. 
Then in MusicRNN you would have `continueSequence(sequence: magenta.INoteSequence, steps: number, temperature?: number, controlArgs?: ControlArgs)`.

How do you pass in the conditioning in your current setup?